dateutil/parser/_parser.py

```diff
@@ -1329,16 +1333,21 @@ def parse(self, tzstr):
                     if not res.stdabbr:
                         offattr = "stdoffset"
                         res.stdabbr = "".join(l[i:j])
+                        for ii in range(j):
+                            used_tokens[ii] = True
                     else:
                         offattr = "dstoffset"
                         res.dstabbr = "".join(l[i:j])
+                        for ii in range(j):
+                            used_tokens[ii] = True
```

Comment:
This loop is done in both branches of this if/else.  Can you de-indent and just do it once?